sparql/hgnc_links.ru

```diff
@@ -0,0 +1,29 @@
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+prefix IAO: <http://purl.obolibrary.org/obo/IAO_>
+prefix MONDO: <http://purl.obolibrary.org/obo/MONDO_>
+prefix RO: <http://purl.obolibrary.org/obo/RO_>
+prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+prefix oio: <http://www.geneontology.org/formats/oboInOwl#>
+prefix def: <http://purl.obolibrary.org/obo/IAO_0000115>
+prefix owl: <http://www.w3.org/2002/07/owl#>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+
+INSERT {
+    ?hgnc rdf:type owl:Class .
```

Comment:
I see that I don't actually know why we need to insert these declarations. Feel free to explain if you want, but if not, that's OK.

Also, I'm wondering why we don't make these declarations when we're building `omim.ttl`. It seems to me that the HGNC information is all coming from `omim.ttl` and not one of these Mondo SSSOM dependencies:
`omim.owl: omim.ttl mondo_exactmatch_omim.sssom.owl mondo_exactmatch_omimps.sssom.owl`

Reply:
> I see that I don't actually know why we need to insert these declarations. Feel free to explain if you want, but if not, that's OK.

This is needed for ROBOT. Basically, without the declarations, the OWL API cant recognise the HGNC identifier as a class, and therefore wont include it in the parsing process.

> Also, I'm wondering why we don't make these declarations when we're building omim.ttl. It seems to me that the HGNC information is all coming from omim.ttl and not one of these Mondo SSSOM dependencies

You are in principle right, but I think this is a bit safer, as you do not need that class declaration when using `skos:exactMatch`; ROBOT can process the mappings as long as they are `AnnotationAssertion()`. You only start needing this when you want to use HGNC identifiers in OWL expressions. So I add it here as a reflex, just to be sure, in case they declaration was not included by the source.

Reply:
Ah OK. I assume there are some OWL expressions that are referencing HGNC then.